client/consensus/aura/src/lib.rs

```diff
@@ -264,11 +272,30 @@ impl<B, C, E, I, P, Error, SO> sc_consensus_slots::SimpleSlotWorker<B> for AuraW
 		StorageChanges<sp_api::TransactionFor<C, B>, B>,
 		Self::Claim,
 		Self::EpochData,
-	) -> sp_consensus::BlockImportParams<B, sp_api::TransactionFor<C, B>> + Send> {
-		Box::new(|header, header_hash, body, storage_changes, pair, _epoch| {
+	) -> Result<sp_consensus::BlockImportParams<B, sp_api::TransactionFor<C, B>>, sp_consensus::Error> + Send + 'static> {
+		let keystore = self.keystore.clone();
+		Box::new(move |header, header_hash, body, storage_changes, pair, _epoch| {
```

Comment:
`pair` here is what is defined as `Self::Claim` which is differently defined in babe and aura. Do you think i should replace it and pass only the public key instead?

Reply:
Here's what i see:

https://github.com/paritytech/substrate/blob/420086de2a535e677f20f805ca4425f656fdf078/client/consensus/slots/src/lib.rs#L212-L215

Then later down the file you'll see:

https://github.com/paritytech/substrate/blob/420086de2a535e677f20f805ca4425f656fdf078/client/consensus/slots/src/lib.rs#L282-L289

Semantically speaking, is "claiming a slot" only signing it? if so, i think this parameter can be removed. Otherwise, i guess it might be a good idea to keep it for now.

What do you think?

Reply:
You apparently only need the public key here. And I don't see any other user that wants the pair, which apparently should not be accessible anymore when it is on the remote key server?

Reply:
done